### PR TITLE
Igamma

### LINKED ISSUE #15712: igamma

### PR DESCRIPTION
**Pull request by selvaraj-sembulingam**
Closes #15712

### CODE REVIEW

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Sorry for the late review. Your PR looks good but can you also add this function for `numpy` backend as well. And if its not too difficult, can you also add it to `paddle` backend.

Thanks

**Comment by selvaraj-sembulingam** [comment]
@sherry30, I couldn't find direct implementations of igamma in numpy and paddle.

In the case of numpy, can I use scipy.special.gammainc ?

Can you provide some guidance on the paddle as well? Do I need to implement igamma using basic paddle functions?

Thanks,
Selvaraj

File: ivy/functional/ivy/experimental/statistical.py [hunk 1]
```diff
@@ -426,3 +426,44 @@ def bincount(
...
+@to_native_arrays_and_back
+def igamma(
+    a: ivy.Array,
+    /,
+    *,
+    x: ivy.Array,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    ivy.Array instance method variant of ivy.igamma. This method simply wraps the
```
Comment (sherry30) [inline ivy/functional/ivy/experimental/statistical.py:443]: The description of the function

**Review by sherry30 — CHANGES_REQUESTED** [review]
You cannot use scipy for numpy's implementation. You'll have to implement it through numpy functions. You could get help from scipy's implementation if it is available on github. 
You can leave the paddle backend for now.

**Review by sherry30 — CHANGES_REQUESTED** [review]
Seems like the numpy backend is failing because there are some differences in the answer with the ground truth. If you think this is because of precision error then feel free to add `rtol_` or `atol_` in the test function. Bu to me it seems like the problem is with  the implementation, I haven't dived too deep into it. In case this is a problem with the implementation, you should try to fix it and let me know if you need help with anything. 

Thanks

**Comment by selvaraj-sembulingam** [comment]
@sherry30 Thanks for pointing this out. There was indeed a minor mistake. Rectified it. The test cases are passing now.

Thanks

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Thanks for fixing the numpy implementation.
After thinking it through, we should also add paddle paddle implementation since the tests run for this backend as well 😅. Sorry for not deciding this sooner but can you add that implementation as well. should be similar to numpy.
and feel free to ask anything if you have any questions as always.

Thanks

**Comment by selvaraj-sembulingam** [comment]
@sherry30 Sure. I have implemented it for the paddle as well. But couldn't able to test it as there are a few config issues in ivy tests.

> ImportError while loading conftest 'E:\Ivy\ivy\ivy_tests\test_ivy\conftest.py'.
> ivy_tests\test_ivy\__init__.py:1: in <module>
>     from .. import config
> E   ImportError: cannot import name 'config' from 'ivy_tests' (E:\Ivy\ivy\ivy_tests\__init__.py)

Thanks

**Review by sherry30 — APPROVED** [review]
Im not sure why you're getting that error, probably some environment error.
It seems to be working fine on my end and all the tests are passing.

Thanks a lot for the contribution and your hard work 😄

**Comment by selvaraj-sembulingam** [comment]
Thanks @sherry30 💯 😄

**Pull request merged**